Ludmila n style 03 contributor ethos article (#191)
InnerSourceCommons/InnerSourceLearningPath: contributor/03-contributor-ethos-article.md

as a visitor you won't be able (or invited) to commit directly to any
code repository.

> Instead, after making your changes to the codebase you'll

Reviewer: I actually liked the conversational part here, it is followed throughout the rest of the articles and, if I remember correctly, referenced to later.

Reviewer 2: Knocking is already mentioned on line 20, so it may be redundant to mention here again.

Reviewer: I agree, closing this one then.

submit them as a pull request. Much like you wouldn't go about making large

> changes and what you consider improvments to your neighbors home, in InnerSource

Reviewer: There's a typo here.
```suggestion
changes and what you consider improvements to your neighbors home, in InnerSource
```

you will anticipate and follow the project's collaboration guidelines. In
turn, your hosts will show you the way - in InnerSource that translates to
existing trusted committers spending their time to mentor guests.

What about those lovely summer parties that you went to? 
There is usually some planning ahead of time to choose the right date and time, to
prepare enough food, or to have it contributed by guests. The same happens for
bigger changes in InnerSource projects: a project will likely ask you to submit 
an issue describing your need and your proposed solution before making a large modification. 
Spending time on initial design instead of
jumping right into the implementation saves contributors from having to
redo a lot of their work. Sharing progress early - even when it's not finished

> yet - helps the host team to mentor the Contributor towards a better solution. Much like

Reviewer: I generally wondered if we could word that directly addressing the reader as in the contributor role. 
We did similar things in the earlier parts of the article I think.

[Yonik's law of unifinished
patches](https://cwiki.apache.org/confluence/display/solr/HowToContribute)
explains: "A half-baked patch in Jira, with no documentation, no tests
and no backwards compatibility is better than no patch at all."

Does that imply that InnerSource projects place no value on face to face
communication? Not quite: there is value in meeting participants face to face.
Remember that all written communication lacks a lot of bandwidth compared to
meeting in person: there are no gestures, no mimics, not even the tone of voice
to help with understanding. In-person meetings are particularly valuable for 
interpersonal challenges, resolving conflicts and misunderstanding. 
However, communication about project decisions should be kept in writing, so that others can
follow along and influence the project, and even years later it's possible
to trace why a certain decision was made.

Here's my general rule of thumb: feel free to meet over coffee. Often that helps 

> build a stronger team. Do make sure though that all decisions are made in a

Reviewer: I did like the example here, since I've seen that effect myself repeatedly if it is not counteracted e.g. using Slack and/or regular contact e.g. over a coffee or the likes.

Reviewer 2: How about referencing multiple locations but letting readers decide what that means for them?

```suggestion
build a stronger team, especially when the team is split into multiple physical locations. Do make sure though that all decisions are made in a
```

> transparent and asynchronous way so that everyone - including those lurking in

Reviewer: We might want to consider replacing or explaining [lurking](https://en.wikipedia.org/wiki/Lurker) here. 
I know what it means, is this common knowledge though?

```suggestion
transparent and asynchronous way so that everyone - including those [lurking](https://en.wikipedia.org/wiki/Lurker) in
```

Reviewer 2: Seems fine to me either way.

sense to offer to fulfill a [30-day
warranty](https://github.com/InnerSourceCommons/InnerSourcePatterns/blob/master/30-day-warranty.md)
on your submission. This can
alleviate the host team's fear of the Contributors not being available for
support with fixing bugs after the time on contribution.

## Anticipate and follow house rules

When you are visiting your neighbors, they will likely help you around in their
apartment: they'll show you the way to their living room and where the restroom 
is located. If you're staying longer, they will probably

> give you more details: in my case an example would be to try not to turn on

Reviewer 2: I like suggestions phrased in the positive.

```suggestion
give you more details: in my case an example would be to avoid turning on
```
